### ISSUE TITLE
"Slash menu": improve instructions and feedback

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#5592]
Follow up to https://github.com/WordPress/gutenberg/issues/5591 to address the second part of the feedback we're getting from assistive technologies users. For full details, please refer to #5591.

Screen reader users are able to use the slash menu. However, only 10 block types are displayed by default and there are no instructions that typing actually filters the list and makes new results appear in the menu.

In the test session reported on #5591 the user had a simple task: add a table with 2 columns, 2 rows and add some data in the cells

The feedback was:
> When I hit slash, table wasn't included in the menu.

I'm wondering if this applies also to sighted users: they can certainly try to type something and they will see the menu changes, but maybe discoverability is not that great.

Unfortunately, the audible message used for screen reader users is always the same for all the auto-completers in Gutenberg. It's always:
`nn results found, use up and down arrow keys to navigate.`

![screenshot 77](https://user-images.githubusercontent.com/1682452/37356185-b27f8110-26e5-11e8-83b3-578b9d70b478.png)

In this specific case, screen reader users need some more context and a meaningful message instructing them to type to refine the search results.

Once they start typing, I think the feedback is good enough. The only issue is the initial message and instructions, clearly not sufficient to understand how this feature works.

![screenshot 78](https://user-images.githubusercontent.com/1682452/37356358-10a60246-26e6-11e8-9954-b992114726dd.png)

Why the initial message "10 results found, use up and down arrow keys to navigate." is confusing?

I guess because users have typed just a slash `/`. Actually, they haven't typed any search term but they get a message that says `10 results found`. Instead, the message should briefly reflect what actually happens in the UI:
- 10 initial choices are displayed
- typing something will display other search redsults

To communicate this, there's the need of a different message.

I'd propose to consider a way to make each autocompleter have its own message. One size doesn't fit all. These auto-completers work  in different ways and they need to be communicated with different, meaningful messages.

Optionally, it would be great to have 2 different messages based on the user workflow:
- a message when the menu opens for the first time
- a different message for the search results

**Comment by nerrad** [comment]
It's been a while since this was first reported.  Needs testing to confirm the issue still exists.

**Comment by afercia** [comment]
The problem still exists. Going to add the "bug" label, as the lack of proper instructions hinders the ability to use this feature.

**Comment by paaljoachim** [comment]
@joedolson Could you take a look and see if this issue is still valid?
Thank you!

**Comment by gwwar** [comment]
This one came up in #core-editor bug triage. https://wordpress.slack.com/archives/C02QB2JS7/p1614280418266100

I think we're using the default component messaging and we need some specific instructions for the inserter usage. Is anyone interested in taking this?
